### ISSUE TITLE
LSP/linting/treesitting support for reStructuredText/Sphinx

### ISSUE THREAD

**Issue by raddevon** [LunarVim/LunarVim#3573]
### Feature Description

Add support for the Sphinx LSP and reStructuredText syntax

### Describe the alternatives you have considered

Manually configuring plugins for support apart from LunarVim

### Support information

- LSP support for Sphinx is available in nvim-lspconfig via [the esbonia language server](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#esbonio).
- The [rstcheck linter](https://github.com/jose-elias-alvarez/null-ls.nvim/blob/main/doc/BUILTINS.md#rstcheck) is avaialable in null-ls.
- nvim-treesitter has `rst` support

**Comment by LostNeophyte** [comment]
from the support information you have provided, it looks like it's already supported

**Comment by LostNeophyte** [comment]
we don't configure linters and lsp servers(there are a few exceptions) out of the box

**Comment by raddevon** [comment]
This must just be a misunderstanding on my part. As I'm using many other languages for the first time, I believe the appropriate LSP is automatically downloaded and installed for me, but that isn't the case with my Sphinx/rst files. That's LunarVim handling that for me, right?

**Comment by LostNeophyte** [comment]
is the filetype `rst` ?

**Comment by LostNeophyte** [comment]
the filetype is skipped, I don't know the reason though
https://github.com/LunarVim/LunarVim/blob/b190f045e896fe5f1646e0807ab0bea008ee8950/lua/lvim/lsp/config.lua#L43
you can create a .config/lvim/ftplugin/rst.lua file with the following content for now 
```
require("lvim.lsp.manager").setup("esbonio")
```

**Comment by LostNeophyte** [comment]
the only other lsp server that has `rst` as its filetype is ltex, but we have it skipped @kylo252 can we remove `rst` from skipped_filetypes?

**Comment by kylo252** [comment]
> the only other lsp server that has `rst` as its filetype is ltex, but we have it skipped @kylo252 can we remove `rst` from skipped_filetypes?

but @raddevon is asking for [esbonio](https://github.com/neovim/nvim-lspconfig/blob/cda7d1d6e068ece255644487c91d9afddd4f87e5/lua/lspconfig/server_configurations/esbonio.lua) 

> the filetype is skipped, I don't know the reason though

similar to `markdown`, any grammar language server should work with `rst`,  which means there are too many servers to _keep_ track of, and we want to avoid that, see https://github.com/LunarVim/LunarVim/pull/2243

> * The [rstcheck linter](https://github.com/jose-elias-alvarez/null-ls.nvim/blob/main/doc/BUILTINS.md?rgh-link-date=2022-12-05T16%3A42%3A33Z#rstcheck) is avaialable in null-ls.

please follow the example config

https://github.com/LunarVim/LunarVim/blob/33e0960ba3fbe4988731a7e297d5f696993f8541/utils/installer/config.example.lua#L85-L92

> * nvim-treesitter has `rst` support

if it's not installed already, just run `:TSInstall rst`

---

please let me know if you encounter any problems!

**Comment by LostNeophyte** [comment]
> > the only other lsp server that has `rst` as its filetype is ltex, but we have it skipped @kylo252 can we remove `rst` from skipped_filetypes?
> 
> but @raddevon is asking for [esbonio](https://github.com/neovim/nvim-lspconfig/blob/cda7d1d6e068ece255644487c91d9afddd4f87e5/lua/lspconfig/server_configurations/esbonio.lua)

I see now that my message wasn't clear at all, I meant that in nvim-lspconfig and [mason the only servers that have `rst` filetype are ltex and esbonio](https://github.com/williamboman/mason-lspconfig.nvim/blob/4674ed145fd0e72c9bfdb32b647f968b221bf2f2/lua/mason-lspconfig/mappings/filetype.lua#L128), so I think removing rst from the skipped_filetypes will only enable esbonio and it only has `rst` in its filetypes so that shouldn't cause us any problems

**Comment by kylo252** [comment]
> I see now that my message wasn't clear at all, I meant that in nvim-lspconfig and [mason the only servers that have `rst` filetype are ltex and esbonio](https://github.com/williamboman/mason-lspconfig.nvim/blob/4674ed145fd0e72c9bfdb32b647f968b221bf2f2/lua/mason-lspconfig/mappings/filetype.lua#L128)

they might be currently missing from lspconfig's filetypes, but will get fixed/added at any point. 

> so I think removing rst from the skipped_filetypes will only enable esbonio and it only has `rst` in its filetypes so that shouldn't cause us any problems

if anything, I wouldn't expect a Sphinx server to show up every time I open an `rst` file, I'd much rather have the grammar server instead.

The whole point is that it's not scalable, and you guys might've missed that you can change the defaults and remove `rst` from that list :nerd_face: 

![image](https://user-images.githubusercontent.com/59826753/206162943-0243cd65-888d-4f31-bfa5-9c5cae8dc7b2.png)

**Comment by LostNeophyte** [comment]
> The whole point is that it's not scalable, and you guys might've missed that you can change the defaults and remove `rst` from that list nerd_face

I've got no idea how I missed that, I thought only skipped servers were configurable

**Comment by raddevon** [comment]
Sorry for the delay. I just got to circle back to this.

I'm not entirely clear what I should do having read through the thread. I believe I got the treesitter installed and the linter enabled, but I'm not clear what I should do on the LSP. Do I need to remove `rst` from skipped filetypes before I can configure an LSP manually? Is it as simple as adding `require("lvim.lsp.manager").setup("esbonio")`, or do I need to also define the file type? Should I manually activate the LSP per buffer somehow since not all `rst` files are Sphinx?

Thank you for your help.
